hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestDataNodeMetrics.java

```diff
@@ -591,14 +592,24 @@ public Boolean get() {
   }
 
   @Test
-  public void testNNRpcMetricsWithNonHA() throws IOException {
+  public void testNNRpcMetricsWithNonHA() throws Exception {
     Configuration conf = new HdfsConfiguration();
     // setting heartbeat interval to 1 hour to prevent bpServiceActor sends
     // heartbeat periodically to NN during running test case, and bpServiceActor
     // only sends heartbeat once after startup
     conf.setTimeDuration(DFS_HEARTBEAT_INTERVAL_KEY, 1, TimeUnit.HOURS);
     MiniDFSCluster cluster = new MiniDFSCluster.Builder(conf).build();
     cluster.waitActive();
+    final MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
```

Comment:
@zhangshuyan0 Thanks for your update. This unit test seems failed based on the yetus reports. Another side, this segment code located at TestNameNodeMetrics should be more proper? Thanks again.

Reply:
Agreed that this isn't the right place. Actually it seems like enhancing `TestRPC#testRpcMetrics()` might be the best place? Or if we can't get it to work there, then I agree `TestNameNodeMetrics` seems better.